providers/nodemailer/src/lib/nodemailer.provider.ts

```diff
@@ -70,10 +60,37 @@ export class NodemailerProvider implements IEmailProvider {
   async checkIntegration(
     options: IEmailOptions
   ): Promise<ICheckIntegrationResponse> {
+    try {
+      const mailData = this.createMailData(options);
+      await this.transports.sendMail(mailData);
+
+      return {
+        success: true,
+        message: 'Integrated successfully!',
+        code: CheckIntegrationResponseEnum.SUCCESS,
+      };
+    } catch (error) {
+      return {
+        success: false,
+        message: error?.message,
+        // nodemailer does not provide a way to distinguish errors
+        code: CheckIntegrationResponseEnum.FAILED,
```

Comment:
I looked through the source code of `nodemailer` to confirm that there is no way to distinguish errors:
https://github.com/nodemailer/nodemailer/blob/v6.4.4/lib/smtp-transport/index.js#L129-L292

Reply:
I think it is okay for now